ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManager.java

```diff
@@ -758,36 +737,20 @@ public List<AmbryReplica> getReplicaIdsByState(AmbryPartition partition, Replica
       Timer.Context operationTimer = helixClusterManagerMetrics.routingTableQueryTime.time();
       long startTime = SystemTime.getInstance().milliseconds();
       Set<AmbryReplica> replicas = new HashSet<>();
-      if (clusterMapConfig.clusterMapUseAggregatedView) {
-        String partitionPath = partition.toPathString();
-        // A partition can be under different resources in different data centers. Due to that, when using aggregated
-        // view, get replicas for all {Resource, partition} pairs.
-        Set<String> resourceNames =
-            globalPartitionToResourceNamesRef.get().getOrDefault(partitionPath, Collections.emptySet());
-        RoutingTableSnapshot globalRoutingTableSnapshot = globalRoutingTableSnapshotRef.get();
-        for (String resourceName : resourceNames) {
-          globalRoutingTableSnapshot.getInstancesForResource(resourceName, partitionPath, state.name())
-              .stream()
-              .map(instanceConfig -> instanceNameToAmbryDataNode.get(instanceConfig.getInstanceName()))
-              .filter(dataNode -> dcName == null || dataNode.getDatacenterName().equals(dcName))
-              .map(dataNode -> ambryDataNodeToAmbryReplicas.get(dataNode).get(partition.toPathString()))
-              .filter(Objects::nonNull)
-              .forEach(replicas::add);
-        }
-      } else {
-        List<String> dcs = dcName != null ? Collections.singletonList(dcName)
-            : dcToDcInfo.values().stream().map(dcInfo -> dcInfo.dcName).collect(Collectors.toList());
-        for (String dc : dcs) {
-          String resourceName = partitionToResourceNameByDc.get(dc).get(partition.toPathString());
-          RoutingTableSnapshot routingTableSnapshot = dcToRoutingTableSnapshotRef.get(dc).get();
-          String partitionPath = partition.toPathString();
-          routingTableSnapshot.getInstancesForResource(resourceName, partitionPath, state.name())
-              .stream()
-              .map(instanceConfig -> instanceNameToAmbryDataNode.get(instanceConfig.getInstanceName()))
-              .map(dataNode -> ambryDataNodeToAmbryReplicas.get(dataNode).get(partition.toPathString()))
-              .filter(Objects::nonNull)
-              .forEach(replicas::add);
-        }
+      List<String> dcs = dcName != null ? Collections.singletonList(dcName)
```

Comment:
nit: what this method really does is return all replica ids, not replica ids by state. We should think about renaming it to make it clear.